geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaConsumerFeatureSource.scala

```diff
@@ -65,7 +67,20 @@ class KafkaConsumerFeatureSource(entry: ContentEntry,
     }
   }
 
-  val features = Maps.newConcurrentMap[String, FeatureHolder]()
+  val cb = CacheBuilder.newBuilder()
+
+  if (expiry) {
+    cb.expireAfterWrite(expirationPeriod, TimeUnit.MILLISECONDS)
+      .removalListener(
+        new RemovalListener[String, FeatureHolder] {
+          def onRemoval(removal: RemovalNotification[String, FeatureHolder]) = {
+            removeFeature(new Delete(removal.getKey))
```

Comment:
Hmm, I don't think this will work - removeFeature only removes from the QT if the feature is still in the cache, which it might not be at this point... You can just remove it directly from the QT here.